app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/PoolContributionAnalysis.kt

```diff
@@ -0,0 +1,90 @@
+package com.babylon.wallet.android.presentation.transaction.analysis
+
+import com.babylon.wallet.android.domain.model.Transferable
+import com.babylon.wallet.android.domain.model.TransferableResource
+import com.babylon.wallet.android.domain.model.assets.PoolUnit
+import com.babylon.wallet.android.domain.model.resources.Pool
+import com.babylon.wallet.android.domain.model.resources.Resource
+import com.babylon.wallet.android.domain.model.resources.metadata.poolUnit
+import com.babylon.wallet.android.presentation.transaction.AccountWithTransferableResources
+import com.babylon.wallet.android.presentation.transaction.PreviewType
+import com.radixdlt.ret.DetailedManifestClass
+import com.radixdlt.ret.ExecutionSummary
+import rdx.works.profile.data.model.pernetwork.Network
+import rdx.works.profile.domain.GetProfileUseCase
+import rdx.works.profile.domain.accountOnCurrentNetwork
+import rdx.works.profile.domain.accountsOnCurrentNetwork
+import timber.log.Timber
+
+suspend fun DetailedManifestClass.PoolContribution.resolve(
+    executionSummary: ExecutionSummary,
+    getProfileUseCase: GetProfileUseCase,
+    resources: List<Resource>,
+    involvedPools: List<Pool>
+): PreviewType {
```

Comment:
You can set the return type to be more explicit like `PreviewType.Transfer.Pool`. The same with the rest of the analysis methods.